src/util/Duration.h

```diff
@@ -0,0 +1,248 @@
+
+  // Construct a `DayTimeDuration` (xsd:DayTimeDuration) object. Implicitly
+  // supported operations (directly on the underlying bits) with
+  // `DayTimeDuration` are `==` and `<=>`. The value range, which is not
+  // affected by normalization, is for each argument: `days: {0 - 1048575}`;
+  // `hours: {0 - 23}`; `minutes: {0 - 59}`; `seconds: {0 - 59.99..}`. Negative
+  // durations are handled over the provided `Type`. If the provided values
+  // for hours, minutes or seconds exceed the typical time limits, they will be
+  // normalized in the sense of conversion to milliseconds by default. Default
+  // duration: `P0DT0H0M0S` (as xsd:dayTimeDuration string).
+  constexpr DayTimeDuration(Type signType = Type::Positive, int days = 0,
+                            int hours = 0, int minutes = 0,
+                            double seconds = 0.00) {
+    AD_CONTRACT_CHECK(days >= 0 && hours >= 0 && minutes >= 0 &&
+                      seconds >= 0.00);
+    int sign = signType == Type::Positive ? 1 : -1;
+    setValues(days, hours, minutes, seconds, sign);
+  }
+
+  // ___________________________________________________________________________
+  // Returns `true` if `DayTimeDuration` w.r.t. this object is `Type::Positive`.
+  constexpr bool isPositive() const {
+    return static_cast<long long>(totalMilliseconds_) >=
+           static_cast<long long>(boundTotalMilliseconds);
+  }
+
+  //____________________________________________________________________________
+  constexpr void setValues(int days, int hours, int minutes, double seconds,
+                           int sign) {
+    AD_CONTRACT_CHECK(sign == 1 || sign == -1);
+    auto totalMilliseconds =
+        static_cast<long long>(days) * dayMultiplier +
+        static_cast<long long>(hours) * hourMultiplier +
+        static_cast<long long>(minutes) * minuteMultiplier +
+        static_cast<long long>(std::round(seconds * secondMultiplier));
+    detail::checkBoundDuration(totalMilliseconds,
+                               static_cast<long long>(boundTotalMilliseconds),
+                               "DayTimeDuration", "xsd:dayTimeDuration");
+    totalMilliseconds_ = totalMilliseconds * sign +
+                         static_cast<long long>(boundTotalMilliseconds);
+  }
+
+  //____________________________________________________________________________
+  // Returns a `DurationValue` struct containing `{days, hours, minutes,
+  // seconds}`.
+  [[nodiscard]] constexpr DurationValue getValues() const {
+    // It is more efficient to extract all values at once from
+    // totalMilliseconds_ w.r.t. days, hours, minutes and seconds because we can
+    // reuse the remainder (updated totalMilliseconds).
+    auto totalMilliseconds = static_cast<long long>(totalMilliseconds_) -
+                             static_cast<long long>(boundTotalMilliseconds);
+    if (!isPositive()) {
+      totalMilliseconds = -totalMilliseconds;
+    }
```

Comment:
This is the issue, they become negative automatically with the previous procedure, but i want them to be positive because the days, hours etc. should be returned as a positive values given the current logic. This is quite confusing/messy code indeed. `std::abs()` can't be used except i remove constexpr w.r.t. the function `getValues()`.

Reply:
`    // Given the current duration logic, the sign of a duration is defined by
    // its Type and not its values, hence the respective values should be always
    // positive (days, hours, minutes, seconds).
    // -> totalMilliseconds range is symmetric to 0, thus if totalMilliseconds <
    // 0, just map it to it's positive value.
    totalMilliseconds =
        totalMilliseconds < 0 ? totalMilliseconds : -totalMilliseconds;`

Reply:
Changing it to:
` totalMilliseconds = totalMilliseconds < 0 ? totalMilliseconds : -totalMilliseconds`
and add the comment:
    // Given the current duration logic, the sign of a duration is defined by
    // its Type and not its values, hence the respective values should be always
    // positive (days, hours, minutes, seconds).
    // -> totalMilliseconds range is symmetric to 0, thus if totalMilliseconds <
    // 0, just map it to it's positive value.